sbol_utilities/helper_functions.py

```diff
@@ -133,3 +133,25 @@ def has_plasmid_role(x):
         return is_plasmid(obj.instance_of.lookup())
     else:
         return False
+
+def is_composite(obj: sbol3.Component) -> bool:
+    """Check if an SBOL Component is a composite.
+
+    :param obj: design to be checked
+    :return: true if composite
+    """
+    def has_dna_type(o):
+        for t in o.types:
+            try:
+                regularized = tyto.SBO.get_uri_by_term(tyto.SBO.get_term_by_uri(t))
```

Comment:
I think this can be done more compactly (and more safely) using tyto's comparison tests.

Reply:
Can you please elaborate on this. Is this a method defined in tyto's library?

Reply:
Check out tyto's inference methods: https://tyto.readthedocs.io/en/latest/index.html#inference